.circleci/config.yml

```diff
@@ -21,14 +21,29 @@ jobs:
       - run:
           name: Install Dependencies
           command: yarn
+      - run:
+          name: Build all code to JavaScript
+          command: yarn build
+      - run: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" >> ~/.npmrc
+      - run:
+          name: Create CLI binaries
+          command: npx pkg --out-path ./cli-binaries ./packages/cli/
```

Comment:
I am assuming you want to test the binaries with the binaries.

I think we can just test the stuff directly from the source code, however if you want to go with the binaries that'll do as well; however we should:

1. Cache the binaries we download from `pgk`, otherwise we're going to spend 2 more minutes every time we run the CI
2. Depending how long these tests take to run, maybe the harness should only run when we merge in `master`
3. Probably the harness should be a separate step.

Reply:
Yep the goal here is definitely to test the actual binaries. We want to make sure the built product is working as expected, as there have been a few issues with the build process in the past.

Reply:
As for branches / master, we can worry about that later. This PR will be putting them everywhere and when it annoys us we can change it.